### PR TITLE
WT-13239 Generate a bazel-compatible analog of dist/filelist

### PR DESCRIPTION
**Pull request by ershov**

### CODE REVIEW

File: dist/filelist.bzl
Comment (dmoody256) [inline dist/filelist.bzl:6]: Package call is reserved for build files which are specific to the directory location, for bazel files which could be loaded anywhere, it should not include Package commands

Comment (ershov) [reply dist/filelist.bzl]: Done.

**Comment by ershov** [comment]
@dmoody256 Let me know if you want a `WT_` prefix to the variables to avoid name clash.

File: dist/s_bazel.py
Comment (marcbutler) [inline dist/s_bazel.py:7]: At least part of the functionality here (returning a list of source files) is available in `dist.py` via `source_files()`.

File: dist/s_bazel.py
Comment (etienneptl) [inline dist/s_bazel.py:21]: Can `filelist.bzl` be a variable?

File: dist/s_bazel.py
Comment (etienneptl) [inline dist/s_bazel.py:25]: Should we tell `s_all` there is an issue and fail?

**Review by etienneptl — APPROVED** [review]
LGTM, pending suggestions and @dmoody256's approval - please make sure MongoDB compiles with those latest changes.

**Review by dmoody256 — APPROVED** [review]

File: dist/s_bazel.py
Comment (ershov) [reply dist/s_bazel.py:25]: It's not an issue - the file didn't exist and we create it.
If file didn't exist or changed, it will be reported and s_all will treat it as error.

**Pull request merged**
